libs/payments/stripe/src/lib/stripe.manager.spec.ts

```diff
@@ -188,6 +192,34 @@ describe('StripeManager', () => {
       );
       expect(result).toEqual(mockResponse);
     });
+
+    it('throws PromotionCodeCouldNotBeAttachedError error if error is StripeInvalidRequestError', async () => {
+      const stripeError = new Stripe.errors.StripeInvalidRequestError({
+        type: 'invalid_request_error',
+      });
```

Comment:
This test is at the wrong layer. If you ask for a subscription that doesn't exist, it would be odd to get an error about not being able to attach a promotion code. Attaching the promotion code is at the server layer, not this manager layer. Moving them would also remove the duplication in logic here, as it would be a `describe` regarding the service method, and these would each be a different way it can fail.